website/source/guides/security/vault-pki-integration.html.md

```diff
@@ -0,0 +1,550 @@
+identity_policies    []
+policies             ["default" "tls-policy"]
+```
+
+Make a note of this token as you will need it in the upcoming steps.
+
+### Step 9: Configure Consul Template on All Nodes
+
+If you are using the AWS environment provided in this guide, you already have
+[Consul Template][consul-template-github] installed on all nodes. If you are
+using your own environment, please make sure Consul Template is installed. You
+can download it [here][ct-download].
+
+Provide the token you created in [Step
+8](#step-8-generate-a-token-based-on-tls-policy) to the Consul Template
+configuration file located at `/etc/consul-template.d/consul-template.hcl`. You
+will also need to specify the [template stanza][ct-template-stanza] so you can
+render each of the following on your nodes at the specified location from the
+provided templates (the actual templates will be provided in the next step):
+
+* Node certificate
+* Node private key
+* CA public certificate
+
+We will also specify the template stanza to create certs and keys for the Nomad
+CLI (which defaults to HTTP but will need to use HTTPS once once TLS is enabled
+in our cluster).
+
+Your `consul-template.hcl` configuration file should look similar to the
+following (you will need to provide this to each node in the cluster):
+
+```
+vault {
+  address      = "http://active.vault.service.consul:8200"
+  token        = "s.xafiYzh7MCMotHLu2d35hepR"
+  grace        = "1s"
+  unwrap_token = false
+  renew_token  = true
+}
+
+syslog {
+  enabled  = true
+  facility = "LOCAL5"
+}
+
+template {
+  source      = "/opt/nomad/templates/agent.crt.tpl"
+  destination = "/opt/nomad/certs/agent.crt"
+  command     = "pkill -SIGHUP nomad"
+}
+
+template {
+  source      = "/opt/nomad/templates/agent.key.tpl"
+  destination = "/opt/nomad/certs/agent.key"
+  command     = "pkill -SIGHUP nomad"
+}
+
+template {
+  source      = "/opt/nomad/templates/ca.crt.tpl"
+  destination = "/opt/nomad/certs/ca.crt"
+  command     = "pkill -SIGHUP nomad"
+}
+
+template {
+  source      = "/opt/nomad/templates/cli.crt.tpl"
+  destination = "/opt/nomad/certs/cli.crt"
+}
+
+template {
+  source      = "/opt/nomad/templates/cli.key.tpl"
+  destination = "/opt/nomad/certs/cli.key"
+}
+```
+You may change the `source` and `destination` options in your
+configuration to a location you prefer. This environment assumes you have
+created a `templates` and `certs` directory in `/opt/nomad`.
+
+!> Note: we have hard coded the token we created into the Consul Template
+configuration file. Although we can avoid this by assigning it to the
+environment variable `VAULT_TOKEN`, this method can still pose a security
+concern. The recommended approach is to securely introduce this token to Consul
+Template. To learn how to accomplish this, see [Secure
+Introduction][secure-introduction].
+
+### Step 10: Create Templates
+
+The following are the templates used by the configuration file we specified in
```

Comment:
A line here to tell folks to explicitly create these templates in their newly created /templates directory would help keep things rock-solid imo.

Reply:
Resolving this and applying action in other feedback regarding re-ordering the guide and creating directories + templates first